src/api-helper.ts

```diff
@@ -0,0 +1,16 @@
+import { Workspace, WorkspaceAgent } from "coder/site/src/api/typesGenerated"
+
+export function extractAgentsAndFolderPath(
+  workspace: Workspace,
+): [agents: WorkspaceAgent[], folderPath: string | undefined] {
+  // TODO: multiple agent support
+  const agents = workspace.latest_build.resources.reduce((acc, resource) => {
+    return acc.concat(resource.agents || [])
+  }, [] as WorkspaceAgent[])
+
+  let folderPath = undefined
+  if (agents.length === 1) {
+    folderPath = agents[0].expanded_directory
+  }
```

Comment:
If there is more than 1 agent, does everything fail? Should we randomly choose one?